src/components/ProblemSolutionsModal.tsx

```diff
@@ -0,0 +1,200 @@
+              <a
+                className="underline text-blue-600 dark:text-blue-400"
+                href="mailto:usacoguide@gmail.com"
+                target="_blank"
+              >
+                usacoguide@gmail.com
+              </a>
+              .
+            </p>
+            <button
+              className="my-4 btn-primary"
+              onClick={() =>
+                firebaseUser ? showSubmitSolutionModal() : signIn()
+              }
+            >
+              {firebaseUser
+                ? 'Submit a Solution'
+                : 'Sign in to submit a solution'}
+            </button>
+            <div className="h-8" />
+            <h4 className="text-lg font-semibold pb-2 mb-4 border-b border-gray-200 dark:border-gray-800">
+              My Solutions
+            </h4>
+            <div className="space-y-6">
+              {currentUserSolutions?.map(submission => (
+                <div key={submission.id}>
+                  <h4 className="mb-2 text-gray-700 dark:text-gray-100">
+                    {submission.language
+                      ? LANGUAGE_LABELS[submission.language]
+                      : 'Unknown Language'}{' '}
+                    ({submission.isPublic ? 'Public' : 'Private'}){' '}
+                    <button
+                      className="hover:underline text-blue-600 dark:text-blue-300"
+                      onClick={() => {
+                        if (
+                          confirm(
+                            'Are you sure you want to delete this submission?'
+                          )
+                        ) {
+                          deleteSolution(submission.id);
+                        }
+                      }}
+                    >
+                      (Delete)
+                    </button>
+                  </h4>
+                  <div className="text-sm leading-normal">
+                    <CodeBlock
+                      className={
+                        submission.language !== 'unknown'
+                          ? `language-${submission.language}`
+                          : undefined
+                      }
+                    >
+                      {submission.solutionCode}
+                    </CodeBlock>
+                  </div>
+                </div>
+              ))}
+              {currentUserSolutions?.length === 0 && (
+                <span>No solutions yet!</span>
+              )}
+            </div>
+            <div className="h-8" />
+            <h4 className="text-lg font-semibold pb-2 mb-4 border-b border-gray-200 dark:border-gray-800">
+              Other Public Solutions
+            </h4>
+            <div className="space-y-6">
+              {publicSolutions?.map(submission => (
+                <div key={submission.id}>
+                  <h4 className="mb-2 text-gray-700 dark:text-gray-100">
+                    {submission.userName ?? 'Unknown User'} -{' '}
```

Comment:
Maybe use `|` instead of `-` to separate info?

Reply:
sure -- tbh the entire UI system needs to be completely overhauled D: